### PR TITLE
Fix the flaky legacy_service_account_token_clean_up_test.

### PR DESCRIPTION
**Pull request by yt2985**
the syncInterval time.

#### Which issue(s) this PR fixes:

Fixes #
The integration test test/integration/serviceaccount/legacy_service_account_token_clean_up_test.go is flaky in the test case "auto created legacy token without pod binding, secret has not been used after tracking". This fix increases the syncInterval time to avoid the token is deleted too early.

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yt2985** [comment]
Hi, @enj, would you please take a look of this fix? Thank you very much!

**Comment by pacoxu** [comment]
> The integration test test/integration/serviceaccount/legacy_service_account_token_clean_up_test.go is flaky in the test case "auto created legacy token without pod binding, secret has not been used after tracking". This fix increases the syncInterval time to avoid the token is deleted too early.

is there a failed job in CI? I cannot find it in https://storage.googleapis.com/k8s-triage/index.html?text=legacy_service_account_token_clean_up_test.

**Comment by enj** [comment]
Based on the linked PR failure logs, I _think_ this flake is caused by the previous iteration of the controller still being active, which causes it to delete the secret _before_ the current iteration starts execution via the `.Step` call.  @yt2985 is that correct?

```
legacy_service_account_token_clean_up_test.go:189: The secret auto-token-without-pod-mounting-b should not be cleaned up, err: secrets "auto-token-without-pod-mounting-b" not found
```

**Comment by yt2985** [comment]
> Based on the linked PR failure logs, I _think_ this flake is caused by the previous iteration of the controller still being active, which causes it to delete the secret _before_ the current iteration starts execution via the `.Step` call. @yt2985 is that correct?
> 
> ```
> legacy_service_account_token_clean_up_test.go:189: The secret auto-token-without-pod-mounting-b should not be cleaned up, err: secrets "auto-token-without-pod-mounting-b" not found
> ```

Not exactly. The controller is initialized before splitting the tests and thus it is used by all the test cases. The flaky is because I used a fake clock in the controller. But the configmap tracked since label and the creation time of the secret are using real time clock. The fake clock time is stepped two days during each test case. When it came to the last flaky test case, the fake clock was 8 days ahead of the secret creation time and configmap tracked since time (and then the secret is determined by the program to be deleted). Previously I tried to increase the syncInterval time to decrease the frequency of calling evaluateSATokens(), but this does not really fix the flaky. Now I remove the getting and checking the existence of secret step for the flaky test case to fix the issue.

> Based on the linked PR failure logs, I _think_ this flake is caused by the previous iteration of the controller still being active, which causes it to delete the secret _before_ the current iteration starts execution via the `.Step` call. @yt2985 is that correct?
> 
> ```
> legacy_service_account_token_clean_up_test.go:189: The secret auto-token-without-pod-mounting-b should not be cleaned up, err: secrets "auto-token-without-pod-mounting-b" not found
> ```

File: test/integration/serviceaccount/legacy_service_account_token_clean_up_test.go [hunk 1]
```diff
@@ -181,14 +181,13 @@ func TestLegacyServiceAccountTokenCleanUp(t *testing.T) {
...
 				}
-			}
-
-			_, err = c.CoreV1().Secrets(myns).Get(context.TODO(), test.secretName, metav1.GetOptions{})
-			if err != nil {
-				if apierrors.IsNotFound(err) {
-					t.Fatalf("The secret %s should not be cleaned up, err: %v", test.secretName, err)
-				} else {
-					t.Fatalf("Failed to get secret %s, err: %v", test.secretName, err)
+				_, err = c.CoreV1().Secrets(myns).Get(context.TODO(), test.secretName, metav1.GetOptions{})
```
Comment (enj) [inline test/integration/serviceaccount/legacy_service_account_token_clean_up_test.go:184]: Moving this `Get` into this `if` statement has no value - the exact same `.Get` is performed on [line 174](https://github.com/kubernetes/kubernetes/blob/06cdf35b5da878489618820798cf1ac2b5f6931f/test/integration/serviceaccount/legacy_service_account_token_clean_up_test.go#LL174C23-L174C105).

**Comment by enj** [comment]
@yt2985 ping

**Comment by yt2985** [comment]
> > @yt2985 ping
> 
> Hi Mo Khan , as my above comment, how do you think if I just remove that step of getting and checking the existence of secret?

I just removed this part to avoid the flaky.

**Comment by enj** [comment]
@yt2985 can you open a follow up PR that makes the `if test.expectCleanedUp && err == nil {` line actually check that the `err` is a `NotFound` error (instead of being happy with any error).

**Comment by yt2985** [comment]
> @yt2985 can you open a follow up PR that makes the `if test.expectCleanedUp && err == nil {` line actually check that the `err` is a `NotFound` error (instead of being happy with any error).

Thank you, @enj , I revised the PR and pushed.

**Comment by pohly** [comment]
It's still flaking when run with "go test -race" on EKS: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116980/pull-kubernetes-integration-eks/1679009390817972224

**Comment by aojea** [comment]
@liggitt published good instruction on how to detect and reproduce flakes, https://gist.github.com/liggitt/6a3a2217fa5f846b52519acfc0ffece0#running-integration-tests-to-reproduce-flakes

Note for reviewers, PRs that address flakes should past the results of running the test with `stress` and the `race`flag

@yt2985 do you mind doing the exercise to demonstrate the race has been totally eliminated with this PR?

**Comment by yt2985** [comment]
Created #119579 to fix the flaky.

**Pull request merged**